include/sonic/dom/handler.h

```diff
@@ -149,16 +150,12 @@ class SAXHandler {
     size_t old = obj.o.next.ofs;
     obj.setLength(pairs, kObject);
     if (pairs) {
-      // Note: shallow copy here, because resource pointer is owned by the node
-      // itself, likely move. But the node from begin to end will never call
-      // dctor, so, we don't need to set null at here. And this is diffrent from
-      // move.
       size_t size = pairs * 2 * sizeof(NodeType);
       void *mem = obj.template containerMalloc<typename NodeType::MemberNode>(
           pairs, *alloc_);
       obj.setChildren(mem);
-      std::memcpy((void *)obj.getObjChildrenFirstUnsafe(), (void *)(&obj + 1),
-                  size);
+      internal::haswell::xmemcpy_16n((void *)obj.getObjChildrenFirstUnsafe(),
```

Comment:
We should not assume `sizeof(NodeType)` is 16x. Pls providing a branch using std::memcpy

Reply:
use a function templated xmemcpy wrapper here, support fallback to std::memcpy